mule-deployer/src/main/java/org/mule/tools/model/anypoint/CloudHubDeployment.java

```diff
@@ -76,4 +79,8 @@ public void setEnvironmentSpecificValues() throws DeploymentException {
     }
 
   }
+
+  public boolean overrideProperties() {
+    return overrideProperties == null ? true : overrideProperties;
```

Comment:
Maybe changing the definition to `protected boolean overrideProperties = true;`  and adding the `defaultValue` to the annotation will spare this logic